test/cpp_interface.cpp

```diff
@@ -211,27 +211,57 @@ TEST(CppInterface, OptimizeMulitplePasses) {
   EXPECT_EQ(expected_text, text);
 }
 
-TEST(CppInterface, OptimizeClearPasses) {
-  const char* original_text =
-      "OpSource GLSL 450\n"
-      "OpDecorate %1 SpecId 1\n"
-      "%bool = OpTypeBool\n"
-      "%1 = OpSpecConstantTrue %bool\n";
-
+TEST(CppInterface, OptimizeSupportPassTokenOperation) {
   SpvTools t(SPV_ENV_UNIVERSAL_1_1);
-  t.SetMessageConsumer(IgnoreMessage);
   std::vector<uint32_t> binary;
-  EXPECT_TRUE(t.Assemble(original_text, &binary));
-
-  Optimizer o(SPV_ENV_UNIVERSAL_1_1);
-  o.RegisterStripDebugInfoPass()
-      .RegisterFreezeSpecConstantValuePass()
-      .ClearRegisteredPasses();
-  EXPECT_TRUE(o.Optimize(&binary));
+  EXPECT_TRUE(t.Assemble("OpSource GLSL 450", &binary));
 
-  std::string text;
-  EXPECT_TRUE(t.Disassemble(binary, &text));
-  EXPECT_EQ(original_text, text);
+  {  // Do nothing with tokens.
+    CreateFreezeSpecConstantValuePass();
+    auto token = CreateUnifyConstantPass();
+  }
+
+  {  // Reassignment
+    auto token = CreateNullPass();
+    token = CreateStripDebugInfoPass();
+
+    EXPECT_TRUE(Optimizer(SPV_ENV_UNIVERSAL_1_1)
+                    .RegisterPass(std::move(token))
+                    .Run(binary.data(), binary.size(), &binary));
```

Comment:
Note you're also testing that the input vector is aliased with the output vector.  That should be called out, maybe even in its own test.